dulwich/tests/compat/server_utils.py

```diff
@@ -237,6 +237,54 @@ def test_fetch_full_depth_into_shallow_clone_from_dulwich(self):
         self.assertEqual([], _get_shallow(clone))
         self.assertReposEqual(clone, self._source_repo)
 
+    def test_fetch_from_dulwich_issue_88_standard(self):
```

Comment:
Do not that I have made changes to `dulwich/tests/test_server.py` that corrects the responses to the expected answers in 145c6ddb80d0eb7f8be03dc0e0c0f5dfa0b75d30 and added further tests commit 3c276c4edceb5243f933dbc38fbe69b4e5ee8c70 to precisely test the behaviour with multiack-detailed and no-done. However reviewing my own code again I think I may have missed one behaviour which I must test now.